pkg/operator/controllers/checker/subnets.go

```diff
@@ -0,0 +1,63 @@
+package checker
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"fmt"
+
+	maoclient "github.com/openshift/machine-api-operator/pkg/generated/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	azureproviderv1beta1 "sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+)
+
+func getSubnetIDs(ctx context.Context, vnetID string, clustercli maoclient.Interface) (masterSubnetID string, workerSubnetIDs []string, err error) {
```

Comment:
+1 on the concept of this function, but I think it would be less brittle if it just returned a list of subnets without trying to differentiate between masters and workers.  This would imply a refactor in the dynamic validation functions, which would I think also make us stronger.  Effectively the dynamic validation functions would just take in a list of subnets without caring what they referred to, and the callers would then be responsible for mapping these back to a specific failing thing.

Reply:
The dynamic validator contract should probably even be that it receives map[string]struct{} - i.e. a *set* of subnets which it should be checking.

Reply:
One thing that get refactored as a part of this is the `path` variable which is typically represented from the RP's standpoint as `properties.workerProfiles[%i].subnetID`.  This doesn't make any sense from context of the operator which won't know about the openshiftcluster document so in `dynamic.go` it was removed.  There is some sorting that was done on the workerProfiles array for consistency, but that no longer is applicable from the operator standpoint.  

Change is made.